### PR TITLE
Increment IP addrress safely

### PR DESCRIPTION
**Pull request by nak3**
To define gateway address, current code increments network
address. However, the code does not care endian and overflow. (Since
it increments network address, overflow should not happen, though.)

This patch introduces incrementIPv4Addr() and increments the address
safely.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by mogren** [comment]
Hmm, unit tests failed.

**Comment by nak3** [comment]
Oh, sorry, it was my mistake. Fixed it.

File: pkg/networkutils/network_test.go [hunk 1]
```diff
@@ -196,6 +196,32 @@ func TestSetupHostNetworkNodePortEnabled(t *testing.T) {
...
+	testCases := []struct {
+		name     string
+		ip       net.IP
+		expected net.IP
+		err      bool
+	}{
+		{"increment", net.IPv4(10, 0, 0, 1), net.IPv4(10, 0, 0, 2).To4(), false},
+		{"carry up 1", net.IPv4(10, 0, 0, 255), net.IPv4(10, 0, 1, 0).To4(), false},
+		{"carry up 2", net.IPv4(10, 0, 255, 255), net.IPv4(10, 1, 0, 0).To4(), false},
+		{"overflow", net.IPv4(255, 255, 255, 255), nil, true},
```
Comment (mogren) [inline pkg/networkutils/network_test.go]: Nice test cases.

File: pkg/networkutils/network.go [hunk 2]
```diff
@@ -507,22 +508,24 @@ func setupENINetwork(eniIP string, eniMAC string, eniTable int, eniSubnetCIDR st
...
 
 	if err != nil {
 		return errors.Wrapf(err, "eni network setup: invalid ipv4 cidr block %s", eniSubnetCIDR)
 	}
 
-	// TODO: big/little endian:  convert subnet to gw
-	gw.IP[3] = gw.IP[3] + 1
+	gw, err := incrementIPv4Addr(ipnet.IP)
+	if err != nil {
+		return errors.Wrapf(err, "eni network setup: faild to define gateway address from %v", ipnet.IP)
```
Comment (mogren) [inline pkg/networkutils/network.go]: Nit: Misspelled `faild`

Comment (nak3) [reply pkg/networkutils/network.go]: Thank you :thumbsup: Fixed it to `failed`

**Pull request merged**